Feature allcategories support

Vadym-Ivanenko opened the pull request.

Vadym-Ivanenko committed:
allcategories js

1.html done
2.css done
3. all categories done

Vadym-Ivanenko committed:
1

Vadym-Ivanenko committed:
support list

created support-list

Vadym-Ivanenko committed:
1

Morifer79 committed:
resolve conflict

Morifer79 committed:
Merge branch 'main' into feature_allcategories-support

Morifer79 commented on src/css/support.css:
```diff
@@ -0,0 +1,169 @@
+    color: rgba(17, 17, 17, 0.60);
+    background-color: transparent;
+
+    transition: color 250ms cubic-bezier(0.4, 0, 0.2, 1)
```
в style.css уже есть переменная для  transition - используй её

Morifer79 commented on src/css/support.css:
```diff
@@ -0,0 +1,169 @@
+@media screen and (min-width: 768px) {
+    .allcategories-list__btn {
+        font-size: 18px;
+        line-height: 24px;
```
line-height указываем в виде коэффициента... в твоём случае это 1.33

Morifer79 commented on src/css/support.css:
```diff
@@ -0,0 +1,169 @@
+
+@media screen and (min-width: 768px) {
+    .allcategories-list__btn:focus {
+        line-height: 24px;
```
то же самое... 
line-height / font-size = коэффициент

Morifer79 commented on src/css/support.css:
```diff
@@ -0,0 +1,169 @@
+
+    border-radius: 16px;
+    background: radial-gradient(189.32% 190.93% at 59.76% -23.42%, #4F2EE8 18.03%, #FFF 100%);
+    box-shadow: 0px 10px 10px 0px rgba(245, 245, 247, 0.10);
```
новый синтаксис...
не критично, но 0.10 записываем просто как .1
и то же самое на строке 71 (я пропустил)

Vadym-Ivanenko replied:
Вибач , не розумію що маєш на увазі, можливо ткрлю. Поясни, що маєш на увазі

Morifer79 replied:
вот как должно быть box-shadow: 0 10px 10px 0 rgba(245, 245, 247, .1);

Morifer79 commented on src/css/support.css:
```diff
@@ -0,0 +1,169 @@
+.support-title {
+    font-size: 22px;
+    font-weight: 700;
+    line-height: 24px;
```
переведи в коэффициент

Morifer79 commented on src/css/support.css:
```diff
@@ -0,0 +1,169 @@
+@media screen and (min-width: 768px) {
+    .support-title {
+        font-size: 24px;
+        line-height: 28px;
```
снова коэффициент...

Morifer79 commented on src/js/allCategories-markup.js:
```diff
@@ -0,0 +1,9 @@
+export default function categoriesListMarkup(data) {
```
в самом начале договаривались, чтобы дефолтных экспортов не было...
применяем только именованные

Morifer79 commented on src/js/allCategories-render.js:
```diff
@@ -0,0 +1,50 @@
+  bookId: 'https://books-backend.p.goit.global/books/',
+};
+
+export default async function getBookAPI(type, value = '') {
```
дефолтные экспорты не применяем...

Morifer79 commented on src/js/support-list.js:
```diff
@@ -0,0 +1,57 @@
+import razom from '../images/04.png';
+import actionAgainstHunger from '../images/05.png';
+import worldVision from '../images/09.png';
+import sergiyPrytula from '../images/06.png';
```
можешь оставить так, но что мы выигрываем потерей 9 строк - я так и не понял)

Vadym-Ivanenko committed:
Merge branch 'main' into feature_allcategories-support

Vadym-Ivanenko committed:
Fix and changes

1. Доробив suppport.html -дописав під блок супорт.
2. Зліквідував allCategories-markup і переніс в Allcategoties-render.
3. Додав svg - arrow-down.svg (не знайшов в спрайті)
4. Закінчив support-render.js
5. Використовую додаткову бібліотеку Swiper. (як заакцептиш мій пулл потрібно буде знову прописати "npm i" , щоб підтягнуло її).
6. В index.js - потрібно буде додати 2 імпорти "support-render" + "allCategories-render"

7. В CSS поправив line-height на коефіцієнти / виправив transition на вже інуючий в проєкті

8. В js виправив дефолтні експорти

Vadym-Ivanenko committed:
Merge branch 'feature_allcategories-support' of https://github.com/Morifer79/code-jedi-project-02 into feature_allcategories-support

Vadym-Ivanenko committed:
Complete and fixed

1. Доробив suppport.html -дописав під блок супорт.
2. Зліквідував allCategories-markup і переніс в Allcategoties-render.
3. Додав svg - arrow-down.svg (не знайшов в спрайті)
4. Закінчив support-render.js
5. Використовую додаткову бібліотеку Swiper. (як заакцептиш мій пулл потрібно буде знову прописати "npm i" , щоб підтягнуло її).
6. В index.js - потрібно буде додати 2 імпорти "support-render" + "allCategories-render"

7. В CSS поправив line-height на коефіцієнти / виправив transition на вже інуючий в проєкті

8. В js виправив дефолтні експорти

Vadym-Ivanenko committed:
Merge branch 'main' into feature_allcategories-support

Vadym-Ivanenko committed:
Complete and fixed

1. Доробив suppport.html -дописав під блок супорт.
2. Зліквідував allCategories-markup і переніс в Allcategoties-render.
3. Додав svg - arrow-down.svg (не знайшов в спрайті)
4. Закінчив support-render.js
5. Використовую додаткову бібліотеку Swiper. (як заакцептиш мій пулл потрібно буде знову прописати "npm i" , щоб підтягнуло її).
6. В index.js - потрібно буде додати 2 імпорти "support-render" + "allCategories-render"

7. В CSS поправив line-height на коефіцієнти / виправив transition на вже інуючий в проєкті

8. В js виправив дефолтні експорти
